src/ring_theory/fractional_ideal.lean

```diff
@@ -103,35 +101,41 @@ variables [algebra R P] [loc : is_localization S P]
 
 instance : has_coe (fractional_ideal S P) (submodule R P) := ⟨λ I, I.val⟩
 
-@[simp] lemma val_eq_coe (I : fractional_ideal S P) : I.val = I := rfl
+section set_like
 
-@[simp, norm_cast] lemma coe_mk (I : submodule R P) (hI : is_fractional S I) :
-  (subtype.mk I hI : submodule R P) = I := rfl
+instance : set_like (fractional_ideal S P) P :=
+{ coe := λ I, I.val,
```

Comment:
```suggestion
{ coe := fractional_ideal.val,
```

Reply:
Good point. Originally I placed this whole section before the `coe` to submodules, but that makes it annoying to state much of what comes below.